framework/src/meshgenerators/PatchMeshGenerator.C

```diff
@@ -0,0 +1,381 @@
+                                  0.5 * (node_positions[i](1) + node_positions[i + 4](1)),
+                                  0.5 * (node_positions[i](2) + node_positions[i + 4](2))});
+
+      // Four on interior hex back face
+      for (unsigned i = 8; i < 12; ++i)
+        node_positions.push_back(
+            {0.5 * (node_positions[i](0) + node_positions[(i + 1) % 4 + 8](0)),
+             0.5 * (node_positions[i](1) + node_positions[(i + 1) % 4 + 8](1)),
+             0.5 * (node_positions[i](2) + node_positions[(i + 1) % 4 + 8](2))});
+
+      // Four on interior hex front face
+      for (unsigned i = 12; i < 16; ++i)
+        node_positions.push_back(
+            {0.5 * (node_positions[i](0) + node_positions[(i + 1) % 4 + 12](0)),
+             0.5 * (node_positions[i](1) + node_positions[(i + 1) % 4 + 12](1)),
+             0.5 * (node_positions[i](2) + node_positions[(i + 1) % 4 + 12](2))});
+
+      // Four on remaining interior hex edges
+      for (unsigned i = 8; i < 12; ++i)
+        node_positions.push_back({0.5 * (node_positions[i](0) + node_positions[i + 4](0)),
+                                  0.5 * (node_positions[i](1) + node_positions[i + 4](1)),
+                                  0.5 * (node_positions[i](2) + node_positions[i + 4](2))});
+
+      // Eight on remaining interior edges
+      for (unsigned i = 0; i < 8; ++i)
+        node_positions.push_back({0.5 * (node_positions[i](0) + node_positions[i + 8](0)),
+                                  0.5 * (node_positions[i](1) + node_positions[i + 8](1)),
+                                  0.5 * (node_positions[i](2) + node_positions[i + 8](2))});
+    }
+
+    for (unsigned node_id = 0; node_id < num_nodes; ++node_id)
+      nodes[node_id] = mesh->add_point(node_positions[node_id], node_id);
+
+    if (_elem_type == "HEX8")
+      makeHex8Elems(*mesh, nodes);
+    else
+      makeHex20Elems(*mesh, nodes);
+
+    boundary_info.sideset_name(1) = "front";
+    boundary_info.sideset_name(2) = "bottom";
+    boundary_info.sideset_name(3) = "left";
+    boundary_info.sideset_name(4) = "right";
+    boundary_info.sideset_name(5) = "top";
+    boundary_info.sideset_name(6) = "back";
+
+    boundary_info.nodeset_name(100) = "bottom_back_left";
+    boundary_info.nodeset_name(101) = "bottom_back_right";
+    boundary_info.nodeset_name(102) = "top_back_right";
+    boundary_info.nodeset_name(103) = "top_back_left";
+    boundary_info.nodeset_name(104) = "bottom_front_left";
+    boundary_info.nodeset_name(105) = "bottom_front_right";
+    boundary_info.nodeset_name(106) = "top_front_right";
+    boundary_info.nodeset_name(107) = "top_front_left";
+  }
+
+  mesh->prepare_for_use();
+
+  return dynamic_pointer_cast<MeshBase>(mesh);
+}
+
+void
+PatchMeshGenerator::makeQuad4Elems(ReplicatedMesh & mesh, const std::vector<Node *> & nodes)
+{
+  BoundaryInfo & boundary_info = mesh.get_boundary_info();
+  std::vector<std::vector<int>> elements{
```

Comment:
"elements" is OK, but libMesh and many others call this "connectivity" or "element_connectivity".